src/org/httpkit/client.clj

```diff
@@ -185,6 +163,22 @@ an SNI-capable one, e.g.:
 
    bind-address))
 
+;; Normally only need one client per application - params can be configured per req
+(defonce default-client
+  (delay
+    (if (utils/java-version>= 8)
+      (do ; Note "Gilardi scenario"
+        (require 'org.httpkit.sni-client)
```

Comment:
@ptaoussanis Doing a `require` and `ns-resolve` in a delay will mess with graalvm. 

It _might_ _work_, but it will blow up the image size unnecessarily. 

If you would allow me, I'd have a better alternative for this, by moving some of the require + resolve to the top level.

Reply:
And on the other hand, why not just drop support for Java 7. Clojure has been doing this since ages now.

Reply:
@borkdude 

> Doing a require and ns-resolve in a delay will mess with graalvm.

Thanks for considering this. Is this something the recently added [tests](https://github.com/http-kit/http-kit/pull/511) should catch? They seem to pass okay with the change? (I can't speak for image size)

> It might work, but it will blow up the image size unnecessarily

Is it possible to briefly explain why (or point me to something I can read?) - would be curious to understand better why this'd trigger a significant image size increase.

> And on the other hand, why not just drop support for Java 7. Clojure has been doing this since ages now.

I wasn't aware of any downside to continued support, and I'm aware of at least a couple orgs that still use http-kit on Java 7 in production.

I'm not opposed to dropping support if support has non-trivial costs (e.g. issues for GraalVM users).

Reply:
The native image tests already require the namespace beforehand, this is why they currently work. If that wouldn't have been done, the tests would likely fail.

As for "image bloat": doing requires and resolves at runtime (other than top level) will increase babasha from 75mb to 120mb. The reason for this I'm not exactly sure on, but it has to do with the reachability analysis that GraalVM does. If it can't decide that something is unused, it will leave things in.

Reply:
@borkdude Assuming the issue for GraalVM is the lazy require+resolve, is [this](https://github.com/http-kit/http-kit/pull/513/commits/be608e85440a9edfab11360b548a7b07f7796dd6) better?

Reply:
Yep, that solves it. Perhaps doing `(some-> (ns-resolve ..) deref)` would be a bit better even, but not sure if that matters.

Reply:
PR updated 👍 Thanks again!